lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSobolIndices.cxx

```diff
@@ -294,63 +294,168 @@ Scalar FunctionalChaosSobolIndices::getSobolTotalIndex(const UnsignedInteger var
+  bool mustInclude;
+  for (UnsignedInteger i = 0; i < size; ++i)
+  {
+    if (coefficientIndices[i] > 0)
+    {
+      const Scalar coefficientI = coefficients(i, 0);
+      Indices multiIndices(enumerateFunction(coefficientIndices[i]));
+      // Compute denominator
+      if (multiIndices.normInfinite() > 0) totalVariance += coefficientI * coefficientI;
+      // Compute numerator
+      if (multiIndices.normInfinite() == 0)
+      {
+        // Exclude the zero indices
+        mustInclude = false;
+      }
+      else
+      {
+        mustInclude = false;
+        for (UnsignedInteger j = 0; j < groupDimension; ++j)
+        {
+          const UnsignedInteger variableIndex = variablesGroup[j];
+          // Check if any variable in the group has a nonzero degree
+          if (multiIndices[variableIndex] > 0) 
+          {
+            mustInclude = true;
+            break;
+          }
+        }
+      }
+      if (mustInclude) covarianceVariables += coefficientI * coefficientI;
+    } // coefficientIndices[i] > 0
+  } // Loop over the coefficients
+  if (totalVariance > 0.0) return covarianceVariables / totalVariance;
+  else return 0.0;
+}
+
+/* Sobol grouped interaction index accessor */
+Scalar FunctionalChaosSobolIndices::getSobolGroupedInteractionIndex(const Indices & variablesGroup,
     const UnsignedInteger marginalIndex) const
 {
-  // Compute the complementary indice
   const UnsignedInteger inputDimension = functionalChaosResult_.getDistribution().getDimension();
-  if (!variableIndices.check(inputDimension)) throw InvalidArgumentException(HERE) << "The variable indices of a Sobol indice must be in the range [0, dim-1] and must be different.";
-  const Indices complementaryVariableIndices = variableIndices.complement(inputDimension);
-  // Compute total index from complementary first index
-  const Scalar complementaryFirstIndex = getSobolGroupedIndex(complementaryVariableIndices, marginalIndex);
-  const Scalar groupTotalIndex = 1.0 - complementaryFirstIndex;
-  return groupTotalIndex;
+  if (!variablesGroup.check(inputDimension)) throw InvalidArgumentException(HERE) << "The variable indices of a Sobol indice must be in the range [0, dim-1] and must be different.";
+  const Sample coefficients(functionalChaosResult_.getCoefficients().getMarginal(marginalIndex));
+  const Indices coefficientIndices(functionalChaosResult_.getIndices());
+  const UnsignedInteger size = coefficients.getSize();
+  Scalar covarianceVariables = 0.0;
+  const EnumerateFunction enumerateFunction(functionalChaosResult_.getOrthogonalBasis().getEnumerateFunction());
+  // Sum the contributions of all the coefficients associated to a basis vector involving only the needed variables
+  Scalar totalVariance = 0.0;
+  bool mustInclude;
+  for (UnsignedInteger i = 0; i < size; ++i)
+  {
+    if (coefficientIndices[i] > 0)
+    {
+      const Scalar coefficientI = coefficients(i, 0);
+      Indices multiIndices(enumerateFunction(coefficientIndices[i]));
```

Comment:
const

Reply:
Ok. I fixed the similar const issue at e008e7f54f23bedd45db149731931b48c32e5608.